rice-middleware/impl/src/main/java/org/kuali/rice/kew/routelog/web/RouteLogAction.java

```diff
@@ -155,7 +157,12 @@ private ActionRequestValue switchActionRequestPositionIfPrimaryDelegatePresent(
     	 */
     	
     	if (!actionRequest.isRoleRequest()) {
-    		List<ActionRequestValue> primaryDelegateRequests = actionRequest.getPrimaryDelegateRequests();
+            /**
+             * KULRICE-14183 : Detaching the action request object so that re-ordering of action requests for primary delegates does not get incorrectly persisted
+             */
+            EntityManager entityManager = GlobalResourceLoader.getService("rice.kew.entityManager");
```

Comment:
@ssaripel , is it possible to change the code to use one of @kellerj's suggestions?